### ISSUE TITLE
Bypassing the "premium only" restriction on the device picker

### ISSUE THREAD

**Issue by motherboard7444** [SpotX-Official/SpotX#93]
## 📝 Adding the ability to cast to Premium devices

1. Unlocking the ability to cast to other devices even when using spotify free

## 📷 Screenshots

![Screenshot](https://user-images.githubusercontent.com/61032704/181395465-4b695709-6b84-4851-bc51-4f7c7430c61c.png)

**Comment by amd64fox** [comment]
I tried to unlock this limitation, but it's not clear yet how stable it works.
Spotify Connect is very buggy out of the box, god only knows how it will work after these changes, need to test this.



To start testing you need

- only for version 1.1.91
- run this line in powershell terminal:
```ps1
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12; iex "& { $((iwr -useb 'https://raw.githubusercontent.com/amd64fox/SpotX/main/Install.ps1').Content) } -confirm_uninstall_ms_spoti -testconnect"
```

**Comment by eshaane** [comment]
This works for me without any issues, Could this be a client based issue more then a patch based?
![Spotify-Screenshot](https://user-images.githubusercontent.com/84348006/182052944-635ded80-c535-4477-aa88-0e73cc6fa609.png)

**Comment by amd64fox** [comment]
> This works for me without any issues, Could this be a client based issue more then a patch based?

they do not restrict all devices, for example, try running Spotify on android

**Comment by eshaane** [comment]
![image](https://user-images.githubusercontent.com/84348006/182270024-1a83f133-5270-48e3-94c8-3f084abf5c63.png)
oh yeah it does show that

**Comment by amd64fox** [comment]
Added a new test incoming parameter `-testconnect` for more convenient testing.
Now for testing, you just need to [run this line from PowerShell](https://github.com/amd64fox/SpotX/issues/93#issuecomment-1199132148)

**Comment by eshaane** [comment]
Alr will test it asap

**Comment by eshaane** [comment]
![Screenshot_186](https://user-images.githubusercontent.com/84348006/182447951-405f2662-5fcb-4ff4-a57b-6d3ff5d959d2.png)

**Comment by eshaane** [comment]
seems like there is an issue

**Comment by eshaane** [comment]
![Screenshot_187](https://user-images.githubusercontent.com/84348006/182448719-046a6490-d6eb-4375-9943-3cfdc8f41cec.png)

**Comment by amd64fox** [comment]
@Gam3rrXD  
You can uninstall Spotify and run spotx installation again

**Comment by eshaane** [comment]
o ight

**Comment by amd64fox** [comment]
Does switching the audio stream to the phone work?

**Comment by eshaane** [comment]
![Screenshot_189](https://user-images.githubusercontent.com/84348006/182452241-32d94aff-2028-47c4-843b-03baf55d2b45.png)
works now but I am unable to use spicetify with tha patch

**Comment by eshaane** [comment]
> Does switching the audio stream to the phone work?

yea that works

**Comment by eshaane** [comment]
the only issue is spicetify

**Comment by eshaane** [comment]
![Screenshot_190](https://user-images.githubusercontent.com/84348006/182452766-3d4f0ad0-2333-4681-bb5d-547289084367.png)
after i install spicetify

**Comment by amd64fox** [comment]
> I am unable to use spicetify with tha patch

Okay, I'll look for the cause.
Thanks for the feedback.

**Comment by eshaane** [comment]
alr thanks :)

**Comment by amd64fox** [comment]
strange, I have it working together with spicetify
maybe you bet something else ? any additions ? 
Or is the installation process of spicetify something special?

**Comment by eshaane** [comment]
All I have is the dribblish theme, when I install spiceify and run the patch it does the error where it can't find contentpicker4 in xpui.js

**Comment by amd64fox** [comment]
what is the dribblish theme and how is it installed?

**Comment by eshaane** [comment]
It's just a spicetify theme which is just put into the themes folder and I update using "spicetify apply"

**Comment by amd64fox** [comment]
do you have a link to this theme?

**Comment by eshaane** [comment]
https://github.com/spicetify/spicetify-themes/tree/master/Dribbblish

**Comment by amd64fox** [comment]
Weird, installed the dribblish theme but the error still doesn't reproduce

and the installation of spicetify is standard through this command?
```ps1
iwr-useb https://raw.githubusercontent.com/spicetify/spicetify-cli/master/install.ps1 | iex
```

**Comment by eshaane** [comment]
Yes I used that command

**Comment by eshaane** [comment]
I am using a custom colors.ini file though

**Comment by amd64fox** [comment]
can you share your `xpui.js` file before you try to install spotx ? 
That would make it easier to understand what the problem is.

**Comment by eshaane** [comment]
alright

**Comment by eshaane** [comment]
https://file.io/gEJLuELGaJaa

**Comment by eshaane** [comment]
its not letting me upload directly to github

**Comment by amd64fox** [comment]
Now it’s clear what’s the matter, you have Spotify version 1.1.90, but you need 1.1.91
I indicated in that [post above](https://github.com/amd64fox/SpotX/issues/93#issuecomment-1199132148) that 1.1.91 is required, but you apparently missed this moment :)

**Comment by eshaane** [comment]
oh alright let me try again

**Comment by eshaane** [comment]
ok so new issue, i install the version, then it works, I install spiceify then this occurs 
![image](https://user-images.githubusercontent.com/84348006/182473927-e0d9aa39-a48a-43a7-b415-bb7f4ae3cfe4.png)

**Comment by amd64fox** [comment]
https://github.com/amd64fox/SpotX/issues/93#issuecomment-1203085858

**Comment by eshaane** [comment]
i did that twice

**Comment by eshaane** [comment]
![image](https://user-images.githubusercontent.com/84348006/182475235-4d1bd4ed-3695-4614-a8a2-3b8a72b2b216.png)
![image](https://user-images.githubusercontent.com/84348006/182475252-9cb0a006-6d60-4e10-819f-f66b01af66ab.png)
![image](https://user-images.githubusercontent.com/84348006/182475277-7ebabbd0-e5c3-4acd-8d4b-263a31c343cc.png)

**Comment by amd64fox** [comment]
spicetify also needs to be completely uninstalled and installed fresh

**Comment by eshaane** [comment]
did that too but will try again

**Comment by eshaane** [comment]
yeah same issue complete uninstall and reinstall

**Comment by amd64fox** [comment]
You have something restoring the old xpui.js file, most likely this is done by spicetify, completely remove the spicetify folders from Local and Roaming, also completely remove Spotfiy itself

You can use [this patch](https://github.com/amd64fox/Uninstall-Spotify) to uninstall Spotify completely

**Comment by eshaane** [comment]
ok i pinpointed the issue, i installed spicetify and it worked 
BUT when i installed dribblish using this command it broke

Invoke-WebRequest -UseBasicParsing "https://raw.githubusercontent.com/spicetify/spicetify-themes/master/Dribbblish/install.ps1" | Invoke-Expression

**Comment by eshaane** [comment]
@amd64fox i found the issue 100%, there is really no issue. After the patch says that i open spotify and wahla it works just fine. The main thing is the patch thinks it didnt work even tho it did.
![image](https://user-images.githubusercontent.com/84348006/182492626-a1069b5c-8dc4-4f2f-919c-eb877e43d2b7.png)

**Comment by amd64fox** [comment]
Let's end here, as this is a completely different problem.
or
Create a new issue, although I'm sure you're missing something with a clean install.

**Comment by eshaane** [comment]
alr

**Comment by eshaane** [comment]
I believe this issue can be closed as it is working without bugs.

**Comment by amd64fox** [comment]
There were no reviews about the work of this feature except you

Do you really think it's worth running this for everyone?

**Comment by eshaane** [comment]
> There were no reviews about the work of this feature except you
> 
> Do you really think it's worth running this for everyone?

No one reported a new issue, which means at the moment there is no one that is experiencing and error and either they don't care or they didn't report. If a new issue arises they can just make a new issue on the repo.

**Comment by amd64fox** [comment]
ok i will add it tomorrow for everyone
the incoming parameter `-testconnect` will be removed

**Comment by eshaane** [comment]
Alr thanks 😊

**Comment by 8bit-coder** [comment]
So, is this in the main patch now?

**Comment by 8bit-coder** [comment]
Actually, I just tried the new patch and I believe there's still some work to be done:

I tried to connect to the same device posted in the original issue(I also have a Yamaha RX-A730) and I clicked connect but spotify would just say connecting and the receiver says "need premium"(image attached of issue):

![image](https://user-images.githubusercontent.com/24598233/184522344-4f68a03e-c959-4b5d-b693-18b38b41d1fe.png)

I believe there's more work to be done in order to trick any device into really thinking that the cast to device feature is unlocked.

**Comment by amd64fox** [comment]
In this case, as I understand it, a premium check is used through the server, so it’s impossible to bypass it here, as well as set a very high quality for listening to music, all these are server functions that cannot be deceived.

But not all "connect" devices do this check.

**Comment by amd64fox** [comment]
added for everyone, let's see the reaction
